### PR TITLE
bug 1490727: implement recurring payment designs

### PR DESCRIPTION
**Pull request by joshJarr**
**THIS PR IS BLOCKED BY https://github.com/mozilla/kuma/pull/5061**
Please merge #5061 first

-----

**Requires `RECURRING_PAYMENT_BETA_FLAG` to be enabled**

Done some refactoring in the styles and included the flow for the recurring payments. The popover banner should not display the new UI and work as usual. The new UI can be accessed on the `/payments` view, with the recurring payments at `/payments/recurring`.
Design is implemented and passed QA.

**In this PR**

- bug 1490727: Updated media queries to tablet and mobile displays fit. (https://trello.com/c/NDTSpBwz/37-update-tablet-breakpoints-for-the-contribution-banner)
         - Updated media queries to tablet and mobile displays fit.
         - Solve margin layout on label inputs

- https://trello.com/c/FArWmeK0/2-as-a-user-i-can-select-either-one-time-or-recurring-so-that-i-can-decide-the-way-i-want-to-pay
 
- https://trello.com/c/fFLk8AEi/4-as-a-user-i-can-choose-the-amount-that-i-want-to-set-for-recurring-payments-so-that-i-can-pay-the-amount-i-can-afford

- https://trello.com/c/iDMBjyk6/9-as-a-user-i-need-to-be-able-to-enter-my-details-for-recurring-payments-so-that-i-can-set-up-my-recurring-payments

- (sprint 2) https://trello.com/c/krA4aQqY/6-as-a-user-i-can-see-the-payment-options-for-recurring-payments-so-that-i-can-see-if-my-payment-option-is-accepted

**Notes for QA**
- [ ] ensure popover works as usual
- [ ] ensure the popover animation works as expected
- [ ] ensure banner displays on the FAQ page correctly when feature is enabled.
- [ ] ensure the FAQ page works as usual when the recurring payment flag is disabled
- [ ] ensure the user cannot hit `/payments/recurring` when the recurring payments flag is disabled
- [ ] ensure the new UI and recurring payments displays when the waffle flag is enabled
- [ ] ensure the recurring payment form looks correctly across browsers and devices
- [ ] ensure recurring payment form validation works
- [ ] ensure recurring payment form can be submitted
- [ ] ensure details are saved in stripe correctly
- [ ] ensure recurring payment tabs redirect the user correctly
- [ ] ensure that being logged into MDN displays your user name and email in the form field by default
- [ ] ensure you can edit your user detail in the payment form if they are pre-filled
- [ ] ensure using stripe's test error cards correctly triggers validation or redirects to an error page upon submission https://stripe.com/docs/testing

**Screenshot of changes**

screen shot 2018-10-31 at 12 03 09
screen shot 2018-10-31 at 12 02 52
screen shot 2018-10-31 at 12 02 40
screen shot 2018-10-31 at 12 02 22

### CODE REVIEW

**Comment by jwhitlock** [comment]
@joshJarr can you rebase this now that PR #5072 is merged?

File: kuma/payments/tests/test_views.py [hunk 1]
```diff
@@ -3,6 +3,15 @@
 
 from kuma.core.tests import assert_no_cache_header
 from kuma.core.urlresolvers import reverse
+from kuma.users.models import User
+
+from django.test.utils import override_settings
+
+
+def create_user():
```
Comment (michal-macioszczyk) [inline kuma/payments/tests/test_views.py]: this should be deleted, from line 6 to 14

**Comment by joshJarr** [comment]
@jwhitlock merged with master. Looks broken on staging, maybe because `RECURRING_PAYMENT_BETA_FLAG` isn't enabled?

File: kuma/static/styles/components/payments/_vars.scss [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+
+// Banner gradient colors
+$payment-background-colors: #83cff3,#ebf8e1;
+$payment-background-gradient: linear-gradient(to right, nth($payment-background-colors, 1) , nth($payment-background-colors, 2));
+
+// Payment banner form varibles
+$payment-form-border-radius: 4px;
+$payment-form-button-color: #3143e2;
+$payment-form-button-width: 190px;
+$payment-form-container-backdrop-color: #ffffff;
```
Comment (schalkneethling) [inline kuma/static/styles/components/payments/_vars.scss]: Nit use shortened form:

```suggestion
$payment-form-container-backdrop-color: #fff;
```

File: kuma/static/styles/components/payments/banner.scss [hunk 3]
```diff
@@ -118,12 +138,75 @@
         box-sizing: border-box;
         margin: 0 auto;
         max-height: 100%;
-        max-width: 1320px;
-        padding: 28px;
+        max-width: $payment-banner-max-width;
+        padding: $payment-banner-padding;
```
Comment (schalkneethling) [inline kuma/static/styles/components/payments/banner.scss]: Kudos on replacing all of these with variables 👍

Comment (joshJarr) [reply kuma/static/styles/components/payments/banner.scss]: Thanks! Slowly trying to refactor this feature so it's a little more manageable.

File: kuma/static/styles/components/payments/_vars.scss (hunk 2 above)
Comment (joshJarr) [reply kuma/static/styles/components/payments/_vars.scss]: 👍

File: kuma/static/js/tooltip.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            var tooltipId = target.get(0).getAttribute('aria-controls');
+            var tooltip = doc.getElementById(tooltipId);
+
+            if (!tooltip) {
+                return;
+            }
+
+            tooltip = $(tooltip);
+
+            if (tooltip.hasClass('is-open')) {
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/static/js/tooltip.js]: 👍 I think we're planning on becoming less dependant on jQuery, will refactor some more when we have time towards the end of the sprint, hope that's okay!

File: kuma/static/js/tooltip.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            }
+
+            var tooltipId = target.get(0).getAttribute('aria-controls');
+            var tooltip = doc.getElementById(tooltipId);
+
+            if (!tooltip) {
+                return;
+            }
+
+            tooltip = $(tooltip);
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/static/js/tooltip.js]: 👍 I think we're planning on becoming less dependant on jQuery, will refactor some more when we have time towards the end of the sprint, hope that's okay!

For now, I've remove this line and explicitly wrapped `$(tooltip)` whenever it's necessary. 
I'll refactor this section further next sprint. 
Thanks!

File: kuma/static/js/tooltip.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+                };
+            }
+
+            $(tooltip).css(position);
+
+            // Switch back to display: none; which will be removed when openTooltip() is called.
+            tooltip.removeClass('is-opening');
+        }
+
+        // Handle clicks to the tooltip button.
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/static/js/tooltip.js]: 👍

File: kuma/static/js/tooltip.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+            // Is opening is added to remove display: none, but still keep the tooltip hidden
+            // so we can get its dimensions and positino without showing it.
+            tooltip.addClass('is-opening');
+
+            var tooltipDomRect = tooltip.get(0).getBoundingClientRect();
+            var elementDomRect = element.get(0).getBoundingClientRect();
+            var position;
+
+            // Checks for available width to the left of the button and positions the tooltip below the button if none
+            // there isn't enough.
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/static/js/tooltip.js]: 👍

File: kuma/static/js/tooltip.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+        /**
+         * Shows the tooltip and sets the relevant element properties.
+         * @param {Element} tooltip Tooltip element to show.
+         */
+        function openTooltip(tooltip) {
+            // Moves tooltip to the end of the body so we can position correctly.
+            tooltip.appendTo('body');
+            tooltip.addClass('is-open')
+                .attr('aria-hidden', false);
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/static/js/tooltip.js]: 👍

File: kuma/static/js/payments-handler.js [hunk 9]
```diff
@@ -192,38 +65,31 @@
...
             key: stripePublicKey.val(),
             locale: 'en',
             name: 'MDN Web Docs',
             description: 'One-time donation',
+            source: function(token) {
+                submitted = true;
+                stripeToken.val(token.id);
+                addDisabledLocaleStorageItem();
+                // Multiply selection to get value in pennies.
+                // Following Stripe's convention so this is comparable with analytics.
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/static/js/payments-handler.js]: 👍

File: kuma/payments/jinja2/payments/includes/payments-form.html [hunk 10]
```diff
@@ -1,114 +1,62 @@
...
-                    </div>
-                </div>
-            </form>
+        <p class="legal-copy">
+            {% trans %}
+                Payments are not tax deductible. Your payment details will be processed by
+                <a href="https://stripe.com/" rel="noopener" target="_blank">Stripe</a>.
+                Mozilla will keep record of your payment in
+                accordance with its
+                <a href="https://www.mozilla.org/en-US/privacy/websites" target="_blank">Privacy Notice</a>,
```
[the remark this answers is not in the document]
Comment (joshJarr) [reply kuma/payments/jinja2/payments/includes/payments-form.html]: 👍

**Comment by joshJarr** [comment]
@schalkneethling Addressed your comments, thanks!

**Pull request merged**